pennylane/io/qasm_interpreter.py

```diff
@@ -107,8 +111,69 @@ def __init__(self, context):
             context["aliases"] = {}
         if "wires" not in context:
             context["wires"] = []
+        if "scopes" not in context:
+            context["scopes"] = {"subroutines": {}}
+        if "wire_map" not in context or context["wire_map"] is None:
+            context["wire_map"] = {}
         self.context = context
 
+    def update(self, update: dict):
+        """
+        Updated the context dict.
+
+        Args:
+            update (dict): The dict containing the updates.
+        """
+        self.context.update(update)
+
+    def init_subroutine_scope(self, node: SubroutineDefinition):
+        """
+        Inits the outer wires list on a sub context.
+        Args:
+            node (SubroutineDefinition): the subroutine definition.
+        """
+
+        # outer scope variables are available to inner scopes... but not vice versa!
+        # names prefixed with outer scope names for specificity
+        self.scopes["subroutines"][node.name.name] = self._init_clause_in_same_namespace(
+            self, f"{self.name}_{node.name.name}"
```

Comment:
This is probably sufficient now that we know there won't be nested scopes.
```suggestion
            self, node.name.name"
```